src/main/java/forestry/api/recipes/IForestryRecipe.java

```diff
@@ -5,5 +5,86 @@
  ******************************************************************************/
 package forestry.api.recipes;
 
-public interface IForestryRecipe {
+import java.util.UUID;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+
+public interface IForestryRecipe extends IRecipe<IInventory> {
+
+	static ResourceLocation anonymous() {
+		return new ResourceLocation("forestry", "anonymous_" + UUID.randomUUID());
+	}
+
+	// <editor-fold desc="Ignore these methods, we just piggy back off Minecraft's system for recipe sync">
+	@Deprecated
+	@Override
+	default boolean matches(IInventory inv, World worldIn) {
+		return false;
+	}
+
+	@Deprecated
+	@Override
+	default ItemStack getCraftingResult(IInventory inv) {
+		return ItemStack.EMPTY;
+	}
+
+	@Deprecated
+	@Override
+	default boolean canFit(int width, int height) {
+		return false;
+	}
+
+	@Deprecated
+	@Override
+	default ItemStack getRecipeOutput() {
+		return ItemStack.EMPTY;
+	}
+
+	@Deprecated
+	@Override
+	default NonNullList<ItemStack> getRemainingItems(IInventory inv) {
+		return NonNullList.create();
+	}
+
+	@Deprecated
+	@Override
+	default NonNullList<Ingredient> getIngredients() {
+		return NonNullList.create();
+	}
+
+	@Deprecated
+	@Override
+	default boolean isDynamic() {
+		return true;
+	}
+
+	@Deprecated
+	@Override
+	default String getGroup() {
+		return "Forestry";
```

Comment:
Are all recipes groups uppercase?

Reply:
Just had a proper look; they are "" if there is no group, or the group is picked from the recipe JSON

Reply:
yea, but all the group names from the jsons are lowercase